### ISSUE TITLE
Feature request: Field to enter comments i.r.o. keys

### ISSUE THREAD

**Issue by Cassady007** [UltimateHackingKeyboard/agent#1293]
Hello – and apologies in advance if I am not doing this correctly.
I am not clear if I should have contacted UHK directly with this query – but did search, and the request does not appear to have been made before.

I am slowly expanding in my use of the UHK >> being able to add additional Keymaps is allowing for major "app-specific" mapping.
My request: 

Currently, with all the permutations in terms of how one can use shortcuts, it is becoming a bit taxing remembering what everything does/triggers.

**I would find it really useful if the Agent Software, allowed one to add a *comment field* to a particular keypress setup.**
It need not be long, but a place where one could quickly add brief notes explaining what that key-combination does when only triggered in App X or Y, would be a nice-to-have addition.
Maybe it's just me, but no harm in making the suggestion!

Thank you for considering!

**Comment by Cassady007** [comment]
Hi – thank you for the quick response. 

Below are two screengrabs, to explain a bit clearer.
**Option A** would involve a further "Comment" tab, that would literally just have an open text field, allowing the user to add comments/notes about that particular keypress, or combination of same – for future reference. 
It wouldn't need to do anything, other than to be saved – so that when the keypress editor is called up, the user can click into the Comments tab, and see any notes made there. 
![UHK_Comment](https://user-images.githubusercontent.com/68002289/86942052-fc1e4300-c144-11ea-9979-2ad37f54daa0.jpg)

**Option B** would be to simply add a space for a comment field at the bottom of the "keypress" editor tab, underneath the "Secondary role" – again, with a simple field to allow some text to be added and saved, as comments for the user. 
![UHK Comment field](https://user-images.githubusercontent.com/68002289/86942263-3851a380-c145-11ea-9e07-a62fcad0e811.jpg)


**WHY?**

I have my FN layer currently allowing for keypresses to trigger F18, F19 and F20.
Over on the Mac, when in Word >> F18 triggers a Keyboard Maestro action that clicks on the "find" button to allow for the next "search item" to be located. F19 triggers an KM action that switches languages for me automatically in Word. F20 is a KM global macro that, when Word is active, brings up a palette that allows me to have my selected text translated by Mate Translate, or does a search of the selected text inside Textexpander. 
Different applications have different actions being fired by different keypresses, in other words. 
Going forward, I intend to set up particular Keymaps that only triggers things in Word, or Scrivener, or Devonthink, or Bookends etc., – and they might be triggering KM macros, or even BetterTouchTool, or Vimac/Shortcat app, or application specific actions.

Now, I *could* make a note of all of these somewhere else (like I do inside KM, or BTT and the like) – but ideally, I would love to be able to make notes inside the Agent – so that I can have them more closely related to the keyboard actions that the UHK will allow me to trigger.
This is more user friendly than me having to remember what FN+F18/F19/F20 does in App A,B or C...

Again, I realise I might be an outlier – but the UHK is *so* powerful, that the above scenario is not difficult for users to implement... 
Having a comments field included, would make this process so much easier. 

Hope that explains things better!
And if not something simple to do, and that there are more pressing issues to resolve (like allowing the triggering of the "Macros" inside the Agent (**yes please!!!!**) – then completely understandable!

**Comment by mondalaci** [comment]
Thanks for elaborating on the above!

I was probably unclear regarding my previous question, so let me rephrase it. How would you expect Agent to render the comments? Would you always click on the key of your interest to see its comment? Or should the key display "Switch language" instead of "Ctrl" without clicking on the key.

**Comment by Cassady007** [comment]
Apologies - yes, I misunderstood. 

My initial thought was that it would be wonderful if the keycap description could be controlled by the user, as in your reply/example.

I just thought it might clash with the aesthetics of the Agent, or (more likely) would be a tricky proposition to fit the text into the varied keycap “sizes” — hence the suggestion of the comment tab, that would be more tucked away! :)

**Comment by mondalaci** [comment]
Gotcha, thanks for explaining! I'm closing this issue, but it'll serve as a useful resource in case we decide to implement it in the future.
